### ISSUE TITLE
Question/doubt about duration

### ISSUE THREAD

**Issue by Franc-Brs** [VROOM-Project/vroom#590]
Hi and thank you for your work, I'm enjoing the project.
I'm currently try to make Vroom work with Ors, I was able to build the containers and if I run `curl -w "%{http_code}" http://localhost:3000/health` I get 200 as it should be, in response.
If I try to run something  like(taken from https://github.com/VROOM-Project/vroom/wiki/Demo-server#usage-example ):

`curl --header "Content-Type:application/json" --data '{"vehicles":[{"id":0,"profile":"driving-car","start":[2.3526,48.8604],"end":[2.3526,48.8604]}],"jobs":[{"id":0,"location":[2.3691,48.8532]},{"id":1,"location":[2.2911,48.8566]}],"options":{"g":true}}' http://localhost:3000/`

I get `{"code":3,"error":"Missing durations."}`. 

I read the API documentation, but I didn't understand where I should put the duration and I didn't understand if i's mandatory.
Is it possibile to have some directions?
Sorry for the basic question, If it's not the right place I'll move or close the issue/ticket.

**Comment by Franc-Brs** [comment]
Thanks for the answer. 
Is "the latest" a valid answer, for ors version? It should be 6.6.0 I can try the re-build everything with another image.
http://localhost:8082/ors/status:

 `{"engine":{"version":"6.6.0","build_date":"2021-10-08T10:35:53Z"},"services":["routing","isochrones","matrix","mapmatching"],"languages":["de","de-de","en","en-us","es","es-es","fr","fr-fr","gr","gr-gr","he","he-il","hu","hu-hu","id","id-id","it","it-it","ja","ja-jp","ne","ne-np","nl","nl-nl","pl","pl-pl","pt","pt-pt","ru","ru-ru","tr","tr-tr","zh","zh-cn"],"profiles":{"profile 1":{"profiles":"driving-car","creation_date":"","storages":{"WayCategory":{},"HeavyVehicle":{},"WaySurfaceType":{},"RoadAccessRestrictions":{"use_for_warnings":"true"}},"limits":{"maximum_distance":100000,"maximum_distance_dynamic_weights":100000,"maximum_distance_avoid_areas":100000,"maximum_waypoints":50}}}} `
 
 I tried to request a matrix taking the example you linked (I changed the points in order to don't go out of bounds)
 
  `  curl -X POST \
  'http:///localhost:8082/ors/v2/matrix/driving-car' \
  -H 'Content-Type: application/json; charset=utf-8' \
  -H 'Accept: application/json, application/geo+json, application/gpx+xml, img/png; charset=utf-8' \
  -d '{"locations":[[12.5786784000,41.9417104000],[12.4468316000,41.9085276000],[12.5209718000,41.8754572000],[12.5170615000,41.8732557000]]}'`

And I get

`{"durations":[[0.0,1619.12,1422.06,1457.53],[1578.39,0.0,1244.37,1274.59],[1347.67,1161.23,0.0,99.24],[1385.56,1098.78,118.71,0.0]],"destinations":[{"location":[12.57862,41.941632],"snapped_distance":10.0},{"location":[12.446727,41.908487],"snapped_distance":9.74},{"location":[12.52091,41.875506],"snapped_distance":7.49},{"location":[12.517172,41.873269],"snapped_distance":9.23}],"sources":[{"location":[12.57862,41.941632],"snapped_distance":10.0},{"location":[12.446727,41.908487],"snapped_distance":9.74},{"location":[12.52091,41.875506],"snapped_distance":7.49},{"location":[12.517172,41.873269],"snapped_distance":9.23}],"metadata":{"attribution":"openrouteservice.org, OpenStreetMap contributors","service":"matrix","timestamp":1633699398460,"query":{"locations":[[12.5786784,41.9417104],[12.4468316,41.9085276],[12.5209718,41.8754572],[12.5170615,41.8732557]],"profile":"driving-car","responseType":"json"},"engine":{"version":"6.6.0","build_date":"2021-10-08T10:35:53Z","graph_date":"2021-10-07T16:21:08Z"}}}`

If I try the same (with curl) for optimization I don0t get anything (but maybe I0m doing something wrong).

**Comment by Franc-Brs** [comment]
> > for some reason your ORS matrix API response doesn't contain durations.
> 
> That may be because the coordinates in your curl command are outside your data extract so ORS raises an error. To check this, you can reuse the curl matrix request above but replacing coordinates with the ones from the `vroom` example payload.

Hi,
do you mean taking data from https://github.com/VROOM-Project/vroom/blob/master/docs/example_3.json ?

  `curl -X POST \
  'http://localhost/ors/v2/matrix/driving-car' \
  -H 'Content-Type: application/json; charset=utf-8' \
  -H 'Accept: application/json, application/geo+json, application/gpx+xml, img/png; charset=utf-8' \
  -d '{"locations":[[2.35044,48.71764],[2.35044,48.71764],[1.98935,48.701],[2.39719,49.07611]]}'`
  
The error "out of bound is raised". using the same file in the in the curl command above i.e:

`curl --header "Content-Type:application/json" --data '{data_from_json_in_the_link}' http://localhost:3000/`

I get the usual `{"code":3,"error":"Missing durations."}`.
I also tried to put the data for which I obtain a matrix in the above curl command but I obtain the same response, i.e. code 3 etc etc.. 

`curl --header "Content-Type:application/json" --data '{"vehicles":[{"id":0,"profile":"driving-car","start":[12.5786784000,41.9417104000],"end":[12.4468316000,41.9085276000]}],"jobs":[{"id":0,"location":[12.5209718000,41.8754572000]},{"id":1,"location":[12.5170615000,41.8732557000]}],"options":{"g":true}}' http://localhost:3000/`

gives me `{"code":3,"error":"Missing durations."}` and:   

`curl -X POST \
  'http://192.168.2.28:8082/ors/optimization' \
  -H 'Content-Type: application/json; charset=utf-8' \
  -H 'Accept: application/json, application/geo+json, application/gpx+xml, img/png; charset=utf-8' \
  -d '{"jobs":[{"id":1,"service":300,"amount":[1],"location":[12.5786784000,41.9417104000],"skills":[1],"time_windows":[[32400,36000]]},{"id":2,"service":300,"amount":[1],"location":[12.4468316000,41.9085276000],"skills":[1]},{"id":3,"service":300,"amount":[1],"location":[12.5209718000,41.8754572000],"skills":[2]},{"id":4,"service":300,"amount":[1],"location":[12.5170615000,41.8732557000],"skills":[2]},{"id":5,"service":300,"amount":[1],"location":[12.5170615000,41.8732557000],"skills":[14]},{"id":6,"service":300,"amount":[1],"location":[12.5170615000,41.8732557000],"skills":[14]}],"vehicles":[{"id":1,"profile":"driving-car","start":[12.5170615000,41.8732557000],"end":[12.5170615000,41.8732557000],"capacity":[4],"skills":[1,14],"time_window":[28800,43200]},{"id":2,"profile":"driving-car","start":[12.5209718000,41.8754572000],"end":[12.5209718000,41.8754572000],"capacity":[4],"skills":[2,14],"time_window":[28800,43200]}]}'`

gives me nothing.

**Comment by Franc-Brs** [comment]
HI, sorry for the delay but this week I had some other non-vroom-reletad things going on. Thanks for the answer by the way. In fact I wasn't able to figure out the problem from the logs (ors) becouse the logs were not much detailed, but maybe I'm not able to understand them or I was looking in the wrong place.

The fact is that probably I've meesed up a little bit with networking in docker and the two container weren't able to talk to each other (I took inspiration from previous ticket https://github.com/VROOM-Project/vroom-docker/issues/27). 
Right now the curl command taken from the example ( https://github.com/VROOM-Project/vroom/wiki/Demo-server#usage-example)

`curl --header "Content-Type:application/json" --data '{"vehicles":[{"id":0,"profile":"driving-car","start":[10.509592,43.855678],"end":[10.519876,43.851584]}],"jobs":[{"id":0,"location":[10.507917,43.853312]},{"id":1,"location":[10.512126, 43.850589]}],"options":{"g":true}}' http://localhost:3000/`

gives me something that make sense.
The call from ors:

   `curl -X POST \
  'http://192.168.2.28:8082/ors/optimization' \ .... etc ....`

Gives me nothing at the moment but I should study its documentation bettere and maybe ask them if I have some more troubles. I think that this ticket can be closed... thanks again!
